modules/cert-manager-acme-http01.adoc

```diff
@@ -0,0 +1,119 @@
+    privateKeySecretRef:
+      name: <secret_for_private_key>                               <2>
+    server: https://acme-staging-v02.api.letsencrypt.org/directory <3>
+    solvers:
+      - http01:
+          ingress:
+            class: openshift-default                               <4>
+----
+<1> Provide a name for the cluster issuer.
+<2> Replace `<secret_private_key>` with the name of secret to store the ACME account private key in.
+<3> Specify the URL to access the ACME server's `directory` endpoint. This example uses the _Let's Encrypt_ staging environment.
+<4> Specify the ingress class. This must be `openshift-default`.
+
+.. Create the `ClusterIssuer` object by running the following command:
++
+[source,terminal]
+----
+$ oc create -f acme-cluster-issuer.yaml
+----
+
+. Create an ingress to expose the service of the user workload.
+
+.. Create a YAML file that defines a `Namespace` object:
++
+.Example `namespace.yaml` file
+[source,yaml]
+----
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: my-ingress-namespace <1>
+----
+<1> Specify the namespace for the ingress.
+
+.. Create the `Namespace` object by running the following command:
++
+[source,terminal]
+----
+$ oc create -f namespace.yaml
+----
+
+.. Create a YAML file that defines the `Ingress` object:
++
+.Example `ingress.yaml` file
+[source,yaml]
+----
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: sample-ingress                                           <1>
+  namespace: my-ingress-namespace                                <2>
+  annotations:
+    cert-manager.io/cluster-issuer: letsencrypt-staging          <3>
+    acme.cert-manager.io/http01-ingress-class: openshift-default <4>
+spec:
+  ingressClassName: openshift-default                            <5>
+  tls:
+   - hosts:
+       - <ingress_domain_name>                                   <6>
+     secretName: sample-tls                                      <7>
+  rules:
+  - host: <ingress_domain_name>                                  <8>
+    http:
+      paths:
+      - path: /
+        pathType: Prefix
+        backend:
+          service:
+            name: sample-workload                                <9>
+            port:
+              number: 80
+----
+<1> Specify the name of the ingress.
+<2> Specify the namespace that you created for the ingress.
+<3> Specify the cluster issuer that you created.
+<4> Specify the ingress class from the cluster issuer.
+<5> Specify the ingress class from the cluster issuer. (TODO: Update this per PR comment)
+<6> Replace `<ingress_domain_name>` with the ingress domain name. You can retrieve this name by running the following command: `oc get ingresses.config/cluster -o jsonpath={.spec.domain}`.
+<7> Specify the secret to store the created certificate in.
+<8> Replace `<ingress_domain_name>` with the ingress domain name. You can retrieve this name by running the following command: `oc get ingresses.config/cluster -o jsonpath={.spec.domain}`.
```

Comment:
Essentially we have,
-  `cluster_base_domain` from `oc get dns.config/cluster -o jsonpath={.spec.baseDomain}`. Eg: `thn-oc412.devcluster.openshift.com` 
- `cluster_ingress_domain` from `oc get ingresses.config/cluster -o jsonpath={.spec.domain}`. Eg: `apps.thn-oc412.devcluster.openshift.com`

And we want to host on the ingress to be <some-new-sub-domain>.apps.<cluster_base_domain> or <some-new-sub-domain>.<cluster_ingress_domain>.

Reply:
@thejasn I updated to use your suggestion, but did we have to make any changes to what the replaceable value is called up in the YAML? Right now, it's `sample.<ingress_domain_name>`.

Reply:
(Related to another comment too, so if you can just look at both of these to make sure the values in the YAML and the replaceable values are correct, and match the descriptions)

Reply:
Okay so here's how I updated them, please let me know if either needs to be adjusted. The line for callout 8 in the YAML is:

```
- host: sample.<hostname>
```

And the description is:

```
<8> Replace `<hostname>` with the hostname. You can use the `<host_name>.<cluster_ingress_domain>` syntax, for example `apps.<cluster_base_domain>`, to take advantage of the `*.<cluster_ingress_domain>` wildcard DNS record and serving certificate for the cluster. Otherwise, you must ensure that there is a DNS record for the chosen hostname.

```

However, both callout 6 and 8 are using `<hostname>`. Just checking that it's expected that these two lines will use the same value. (If not, then we should update the name of the replaceable value of one of them to be different - let me know.)

Reply:
[This](https://github.com/openshift/openshift-docs/pull/56754/files#r1141810259) should answer one your queries. As for having the same value `<hostname>`, that is as expected.

Reply:
Okay, I am removing the "sample." part now then